### DISCUSSION TITLE
Welcome to OpenNept4une Discussions!

### DISCUSSION THREAD

**Discussion by halfmanbear** [OpenNeptune3D/OpenNept4une#19] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by khusmann** [comment]
Hey, thanks for opening this, the discord threat was getting quite crowded :)

My intro: I've been 3dprinting little hobby projects for a year or two now, but through a library printing service -- the N4Pro is the first printer I've owned & have needed to configure myself. A big part of my choosing the N4Pro was finding this project!

So right now I'm just learning everything I can, and will hopefully be more of a contributor down the road... In the meantime I can test things and contribute to docs here and there if that'd be helpful.

**Comment by QuackenAudio** [comment]
I  followed your guide doing the 2nd method with the script
I got to the point where I copy the printer.cfg to the machine, but I cant connect via web gui thru fluidd
so I tried a sudo reboot, and it just told me bad gateway 501 at the top of the page
so I did a physical reboot and now it wont connect at all
no loner see an IP address thru wifi or thru cat5 port on side
before reboot, both worked

is there any other way to get into it? or is it bricked and I have to get  a new board from elegoo somehow?

**Reply by halfmanbear** [reply]
You can connect to the printer via serial (usb C). Are you sure you selected the correct version/ machine?

Worst case you'll just need a eMMC flasher to flash the proper OpenNept4une image.

There is a reason for the warning on that script, I'm not in control of how Elegoo change the image they provide. It did work well but they my change things.

**Reply by QuackenAudio** [reply]
yeah... I think elegoo changed something

I can see it by usb-c via serial in putty
but if I do "sudo ifconfig" it just shows me the loopback only and no other interfaces
"lsusb" also shows nothing plugged into USB even when a drive is physically there
and the USB doesnt seem to be providing any power to anything currently.....
so...something somewhere is unhappy

in the meantime
I've ordered the emmc usb adapter and I also ordered a 2nd mainboard with a chip already loaded on it just in case its a hardware failure and not related to the script...cant hurt to have spare parts anyways

**Reply by halfmanbear** [reply]
It sounds like the wrong board revision was selected (copy's different depending on version /boot/dtb/rockchip/rk3328-roc-cc.dtb) if the wrong one is selected usually it misconfigures ethernet, what machine do you have?

**Reply by QuackenAudio** [reply]
I have a Neptune 4 Max
I did the SSH version not the emmc version

fyi...my board says V2.0
your emmc flash version only shosws v1.0 or v1.1

so what do I need to change to make the ethernet and USB work again?

**Reply by halfmanbear** [reply]
In releases I have this that works with V2.0 max.

Neptune4-Plus+Max-v0.1.5-ZNP-K1-Vx.x.img.xz

If you use serial and run

~/OpenNept4une/OpenNept4une.sh

Select install printer.cfg

Then choose your max and (if or when it asks to install the DTB click yes and reboot)

Depending when you last ran the script it may or may not explicitly ask you (but will do it anyway)

If that still doesn't work I can send you the OG file from elegoo to restore

**Reply by QuackenAudio** [reply]
the question would be, how do I get this onto the machine without USB or ethernet connection
I dont even get power to the USB right now...
is there a way to transfer a file thru the serial usb-c connection?

**Reply by halfmanbear** [reply]
if you select all and copy from here:

[n4plus-n4max-v1.1-2.0-dts.txt](https://raw.githubusercontent.com/halfmanbear/OpenNept4une/main/dtb/n4plus-n4max-v1.1-2.0/n4plus-n4max-v1.1-2.0-dts.txt)

then when your logged in via serial run,

cd ~/ && nano n4plus-n4max-v1.1-2.0-dts.txt

paste the contents of the link in here then quit and save the file with,

control + x 
y
enter

then you want to run (paste as one command),

dtc -I dts -O dtb -o rk3328-roc-cc.dtb n4plus-n4max-v1.1-2.0-dts.txt && sudo cp rk3328-roc-cc.dtb /boot/dtb/rockchip/ && sudo cp rk3328-roc-cc.dtb /boot/dtb-5.16.20-rockchip64/rockchip/overlay/

then 

sudo reboot

**Reply by tyhouck92-prog** [reply]
Hi halfmanbear, I am running into the same issue as the OP where I cannt use my USB/USB-c port after updating to Opennept4une. I tried using the link to the txt file above but It no longer works. is there anyway to get working link?

**Comment by QuackenAudio** [comment]
ok... 
your fix file got me back all the connections of wifi/cat5/usb

but was having some odd issues with files failing permissions when trying to install klipper
used git clone to download the opennept4une stuff and kiauh
I manually deleted all the other stuff from my mks directory and tried again

ran the de_elegoo script in opennept4une

I get this error when I get to installing klipper

###### Creating folder '/home/mks/printer_data/backup' ...
mkdir: cannot create directory ‘/home/mks/printer_data/backup’: Permission denied
and basically at every single folder it tries to create after  that as well

I can manually start adding them, but its a pain in the butt

then when I finally get to the point where I run the openneptu4une.sh file and try to install the printer.cfg
it gives me an error that says

Configuring for Neptune4 Max...
Backup of 'printer.cfg' created as 'backup-printer.cfg.bak'.
Error: Source file /home/mks/OpenNept4une/configs/neptune4max/printer.cfg not found.

I think its just labeled wrong as I found one in 
printer-confs/n4max

```

**Comment by QuackenAudio** [comment]
ok... a little more digging as I took the n4max printer cfg file, and copied it over and renamed it

found that moonraker would not connect...
digging into moonraker logs, a bunch of errors saying cant access files due to permission issues...
so I did a bunch of chmod to fix the files that were having issues


but it doesnt seem to help


-------------------- Log Start | Sat Jan 20 04:56:55 2024 --------------------
data_path: /home/mks/printer_data
is_default_data_path: False
config_file: /home/mks/printer_data/config/moonraker.conf
startup_warnings: []
verbose: False
debug: False
asyncio_debug: False
is_backup_config: False
is_python_package: False
instance_uuid: f2e4bc29fd8a457ea30a6570f75bd797
unix_socket_path: /home/mks/printer_data/comms/moonraker.sock
software_version: v0.8.0-267-g66de18f
git_branch: master
git_remote: origin
git_repo_url: https://github.com/Arksine/moonraker.git
modified_files: []
unofficial_components: []
log_file: /home/mks/printer_data/logs/moonraker.log
python_version: 3.7.3 (default, Oct 11 2023, 09:51:27)  [GCC 8.3.0]
launch_args: /home/mks/moonraker-env/bin/python /home/mks/moonraker/moonraker/moonraker.py -d /home/mks/printer_data
msgspec_enabled: False
uvloop_enabled: False
2024-01-19 22:56:55,253 [confighelper.py:read_file()] - Configuration File '/home/mks/printer_data/config/moonraker.conf' parsed, total size: 466 B
2024-01-19 22:56:55,254 [server.py:add_log_rollover_item()] - 
#################### Moonraker Configuration ####################

[server]
host = 0.0.0.0
port = 7125
klippy_uds_address = /home/mks/printer_data/comms/klippy.sock

[authorization]
trusted_clients = 
	10.0.0.0/8
	127.0.0.0/8
	169.254.0.0/16
	172.16.0.0/12
	192.168.0.0/16
	192.168.1.0/24
	FE80::/10
	::1/128
cors_domains = 
	*.lan
	*.local
	*://localhost
	*://localhost:*
	*://my.mainsail.xyz
	*://app.fluidd.xyz

[octoprint_compat]

[history]

[update_manager]
channel = dev
refresh_interval = 168

#################################################################
All Configuration Files:
/home/mks/printer_data/config/moonraker.conf
#################################################################
2024-01-19 22:56:55,453 [server.py:load_component()] - Component (secrets) loaded
2024-01-19 22:56:55,466 [server.py:load_component()] - Component (template) loaded
2024-01-19 22:56:55,482 [server.py:load_component()] - Component (klippy_connection) loaded
2024-01-19 22:56:55,770 [server.py:load_component()] - Component (application) loaded
2024-01-19 22:56:55,810 [server.py:load_component()] - Component (websockets) loaded
2024-01-19 22:56:55,903 [server.py:load_component()] - Component (dbus_manager) loaded
2024-01-19 22:56:55,914 [server.py:load_component()] - Unable to load component: (database)
Traceback (most recent call last):
  File "/home/mks/moonraker/moonraker/server.py", line 291, in load_component
    component = load_func(config)
  File "/home/mks/moonraker/moonraker/components/database.py", line 956, in load_component
    return MoonrakerDatabase(config)
  File "/home/mks/moonraker/moonraker/components/database.py", line 102, in __init__
    self.database_path, map_size=MAX_DB_SIZE, max_dbs=MAX_NAMESPACES
lmdb.ReadonlyError: /home/mks/printer_data/database: Permission denied
2024-01-19 22:56:55,915 [server.py:main()] - Moonraker Error
Traceback (most recent call last):
  File "/home/mks/moonraker/moonraker/server.py", line 647, in main
    server.load_components()
  File "/home/mks/moonraker/moonraker/server.py", line 255, in load_components
    self.load_component(config, component)
  File "/home/mks/moonraker/moonraker/server.py", line 291, in load_component
    component = load_func(config)
  File "/home/mks/moonraker/moonraker/components/database.py", line 956, in load_component
    return MoonrakerDatabase(config)
  File "/home/mks/moonraker/moonraker/components/database.py", line 102, in __init__
    self.database_path, map_size=MAX_DB_SIZE, max_dbs=MAX_NAMESPACES
lmdb.ReadonlyError: /home/mks/printer_data/database: Permission denied
2024-01-19 22:56:55,916 [server.py:main()] - Server Shutdown





with this error .....   lmdb.ReadonlyError: /home/mks/printer_data/database: Permission denied
I'm not sure what to do.. I've done chmod 0777 for both the database files and the directory and I still get this error

**Reply by halfmanbear** [reply]
You sure you've always run the script as user mks and put the sudo password in when prompted?

Best thing would be to do

cd ~/

sudo rm -rf .

Then git clone the repo again. I made some changes to the script yesterday but it doesn't explain your permissions errors.

I ran it through myself on an elegoo stock image.

**Reply by halfmanbear** [reply]
Try the above if still no luck could be this 

https://github.com/Arksine/moonraker/issues/516

**Comment by QuackenAudio** [comment]
[moonraker.log](https://github.com/halfmanbear/OpenNept4une/files/13996879/moonraker.log)

**Comment by QuackenAudio** [comment]
went back and tried to do everything from scratch as the moonraker link didnt fix anything
so I deleted all directories and files in my home/mks
re downloaded the open neptune git
ran the openenpt4une.sh
went into advanced, chose method 2
went thru full install
appeared to go with zero errors

and the open neptune script said the goal is to update the printer.cfg and to update the Virtual MCU rpi Firmware
so I manually copied the N4M printer cfg file and renamed it to printer.cfg
opened it to verify things

then I followed the printer.cfg instructions about the make stm file 

and I get errors writing the virtual stm thing

doesnt matter if I choose enable low level or not, same result either way

![image](https://github.com/halfmanbear/OpenNept4une/assets/51548286/df2856cb-b72e-44e7-812e-af9d4e12e320)


![image](https://github.com/halfmanbear/OpenNept4une/assets/51548286/afae36a9-fb8e-4d38-b30d-96d404171ce6)



![image](https://github.com/halfmanbear/OpenNept4une/assets/51548286/067dacf7-04d5-4c1b-9e30-86309dfce6ba)

**Reply by halfmanbear** [reply]
Yes but now you need to update the second virtual MCU, via the script or with

https://www.klipper3d.org/RPi_microcontroller.html

**Comment by QuackenAudio** [comment]
I'm sorry I dont understand....
I though I'm supposed to compile it in the open neptune script, then push it via boot button and pin hop

where are the full step by step instructions to do all of this.....

there really needs to be a step one, step 2, step 3 document of some sort

I just give up at this point.... instructions are just not clear enough to make this work from command line

I'm computer savy, write scripts in python, do lots of things with linux...and I'm struggling thru this without an emmc reader/writer


I've ordered a writer, it wont show up for a few weeks because nobody in the US seems to stock any of them (I'd pay next day air shipping to get one right now if I could get it to show up immediately)
I already ordered a backup board with emmc from 3d jake just so I can get this machine back up and running

if theres a way to load all of this or to restore the Elegoo firmware from USB... I'll take it... I'd pay for it... I'll buy you more than just coffee...LOL
but even the elegoo restore they sent me doesnt seem to load at boot...have no idea why its not loading usb at boot

**Reply by halfmanbear** [reply]
Just got your message unfortunately I'm in the UK so cant send an eMMC! The step by step is on the main but it's for flashing eMMC I don't recommend the other method for a reason. I do have an old step by step guide here (go half way in to method 2)

https://docs.google.com/document/d/13q216pEYNhfZm7YI-mBvdqrI1OCaJUwDzQpmW-cM3dc/edit?usp=drivesdk

**Comment by QuackenAudio** [comment]
do you have a link to an image file to write to an emmc for the v2.0 board in my neptune 4 max?

**Reply by halfmanbear** [reply]
All the information is on the readme instructions on the main page if this GitHub even a link to the release images, you will need this one

https://github.com/halfmanbear/OpenNept4une/releases/download/v0.1.5/Neptune4-Plus+Max-v0.1.5-ZNP-K1-Vx.x.img.xz
